src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/document/DocumentManagementServiceTest.java

```diff
@@ -0,0 +1,183 @@
+
+    private ArgumentCaptor<HttpEntity> httpEntityReqEntity;
+
+    @Mock
+    private AuthTokenGenerator authTokenGenerator;
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    @Before
+    public void setUp() {
+        when(authTokenGenerator.generate()).thenReturn(AUTH_HEADER);
+
+        this.httpEntityReqEntity = ArgumentCaptor.forClass(HttpEntity.class);
+
+        documentManagementService = new DocumentManagementService(
+            authTokenGenerator,
+            restTemplate,
+            "http://localhost:8080"
+        );
+    }
+
+    @Test
+    public void should_return_upload_response_with_document_urls_when_docs_are_successfully_uploaded()
+        throws Exception {
+        //Given
+        byte[] test1PdfBytes = toByteArray(getResource("test1.pdf"));
+        byte[] test2PdfBytes = toByteArray(getResource("test2.pdf"));
+
+        Pdf pdf1 = new Pdf("test1.pdf", test1PdfBytes);
+        Pdf pdf2 = new Pdf("test2.pdf", test2PdfBytes);
+
+        when(restTemplate.postForObject(
+            eq("http://localhost:8080/documents"),
+            httpEntityReqEntity.capture(),
+            any()
+        )).thenReturn(getResponse());
+
+        //when
+        List<FileUploadResponse> actualUploadResponse = documentManagementService.uploadDocuments(asList(pdf1, pdf2));
+
+        //then
+        assertThat(actualUploadResponse)
+            .extracting("fileUrl")
+            .containsExactly(
+                "http://localhost:8080/documents/1971cadc-9f79-4e1d-9033-84543bbbbc1d",
+                "http://localhost:8080/documents/0fa1ab60-f836-43aa-8c65-b07cc9bebcbe"
+            );
+
+        assertThat(actualUploadResponse)
+            .extracting("fileName")
+            .containsExactly(
+                "test1.pdf", "test2.pdf"
+            );
+
+        verify(authTokenGenerator).generate();
+        verify(restTemplate).postForObject(
+            eq("http://localhost:8080/documents"),
+            httpEntityReqEntity.capture(),
+            any()
+        );
+    }
+
+    @Test
+    public void should_throw_client_exception_when_service_auth_throws_unauthorized_exception() throws Exception {
+        //Given
+        byte[] test1PdfBytes = toByteArray(getResource("test1.pdf"));
+        byte[] test2PdfBytes = toByteArray(getResource("test2.pdf"));
+
+        Pdf pdf1 = new Pdf("test1.pdf", test1PdfBytes);
+        Pdf pdf2 = new Pdf("test2.pdf", test2PdfBytes);
+
+        when(authTokenGenerator.generate()).thenThrow(new HttpClientErrorException(HttpStatus.UNAUTHORIZED));
+
+        //when
+        Throwable exc = catchThrowable(() -> documentManagementService.uploadDocuments(asList(pdf1, pdf2)));
+
+        //then
+        assertThat(exc)
+            .isInstanceOf(HttpClientErrorException.class);
+
+        verify(authTokenGenerator).generate();
+    }
+
+    @Test
+    public void should_throw_unable_to_upload_doc_exception_when_bulk_scan_service_is_not_whitelisted_in_doc_storage()
```

Comment:
regarding 'when_bulk_scan_service_is_not_whitelisted_in_doc_storage', we don't really know/care what happened on document storage.
All we know is that we got `HttpClientErrorException`.